ad_type_classification/update_ad_types.py

```diff
@@ -0,0 +1,173 @@
+"""Module to classify ad types via precomputed models and update the types in the database.
+"""
+from collections import defaultdict
+import json
+import logging
+import os.path
+import re
+import sys
+from urllib import parse
+
+import joblib
+import pandas
+
+import config_utils
+import db_functions
+from ad_type_classification import ad_creative_body_processor
+
+DATA_DIR = os.path.join('ad_type_classification', 'data')
+URL_RE = re.compile(
+        r'http[s]?://(?:[a-zA-Z]|[0-9]|[$\-_@.&+]|[!*\(\),]|(?:%[0-9a-fA-F][0-9a-fA-F]))+')
+
+
+def _get_canonical_url(url):
+    """Parse string for URL, and strip www. if present.
+
+    Args:
+        url: str of URL to parse.
+    Returns:
+        str of URL (with www. stripped if present) of parsed URL. empty string if parsing failed.
+    """
+    try:
+        parsed_url = parse.urlparse(str(url))
+        domain = str(parsed_url.netloc).lower()
+        path = str(parsed_url.path).lower()
+        if domain.startswith('www.'):
+            domain = domain[4:]
+        return domain + path
+    except ValueError as err:
+        logging.info('%s error while processing url: %s', err, url)
+        return ''
+
+
+def get_normalized_creative_url(row):
+    """Get URL from ad creative data. Either link caption or parsed from creative body.
+
+    Args:
+        row: dict with keys 'ad_creative_link_caption' and 'ad_creative_body'.
+    Returns:
+        string: domain (ie secure.actblue.com) of link in ad creative.
+    """
+    url = ''
+    if row.get('ad_creative_link_caption'):
+        url = row['ad_creative_link_caption']
+    else:
+        match = re.search(URL_RE, row['ad_creative_body'])
+        if match:
+            url = _get_canonical_url(match.group(0))
+    return url
+
+
+def get_canonical_url_to_ad_type_table(canonical_url_to_type_csv_path):
```

Comment:
We could simplify this function if we sanitized  the urls in ad_url_to_type.csv

Reply:
done. I don't think any changes are need in urls in ad_url_to_type.csv